### DISCUSSION TITLE
Interoperability with existing `FORCE_COLOR` uses

### DISCUSSION THREAD

**Discussion by epage** [donatj/force-color.org#98] [General]
Just found out about this project (see astral-sh/uv#3955) which seems to be trying to document and standardize `FORCE_COLOR`s behavior.

It seems like it would be helpful to specify the behavior to be compatible with major users.

Behaviors I've seen through a quick survey
- `FORCE_COLOR` is present and non-empty
  - https://force-color.org/
- `FORCE_COLOR` is present
  - *(Python)* https://docs.python.org/3.13/using/cmdline.html#controlling-color
  - *(Python)* https://github.com/Textualize/rich/blob/master/rich/console.py#L952-L956
  - *(Python)* https://github.com/wntrblm/nox/blob/main/nox/_options.py#L605
  - *(Python)* https://github.com/pytest-dev/pytest/blob/97ed533f63d5780a05702a711555cb6744247a37/src/_pytest/_io/terminalwriter.py#L26-L37
  - *(Python)* https://github.com/pypa/build/blob/main/src/build/__main__.py#L46-L53
- `FORCE_COLOR` is present and bool-ish
  - *(Python)* https://github.com/tox-dev/tox/blob/main/src/tox/config/cli/parser.py#L352-L361
  - *(Python)* https://github.com/python/mypy/blob/master/mypy/util.py#L552-L557
  - *(JS)* https://github.com/chalk/supports-color/blob/main/index.js#L33-L57
    - Note that higher numbers express certain capabilities of the terminal
  - *(rust)* https://github.com/zkat/supports-color/blob/d5c906d1fcd91cd1bf92d22f3d0cca452de8ca9f/src/lib.rs#L38-L43 (modeled after chalk.supports-color)
    - Note that higher numbers express certain capabilities of the terminal

**Comment by shadowspawn** [comment]
For reference, two major implementations I have looked at in node ecosystem and linked from force-color.org do not follow the description on force-color.org. The primary use in both cases is levels 0-3 with some other values explicitly supported.  I think in practice other environment variables like [CLICOLOR_FORCE](https://bixense.com/clicolors/) and [NO_COLOR](https://no-color.org) have simpler patterns of usage than `FORCE_COLOR` for forcing color on/off.

The node implementation mentions being inspired by the `supports_color` package. Levels 1-3 correspond to 16 colours, 256 colours, and 16M colours.

| FORCE_COLOR | result  |
|-----------------|-------|
| [supports_color](https://github.com/chalk/supports-color/blob/c214314a14bcb174b12b3014b2b0a8de375029ae/index.js#L33) |
| empty | on, level 1 |
| '0' | off |
| '1', '2', '3' | on |
| 'true' | on, level 1 |
| 'false' | off |
| other non-number | ignored |
| [node](https://github.com/nodejs/node/blob/0a00217a5f67ef4a22384cfc80eb6dd9a917fdc1/lib/internal/tty.js#L109) |
| empty | on, level 1 |
| '1', '2', '3' | on |
| 'true' | on, level 1 | 
| other | off |
